tests/boost/dataset_broker_producer2.cpp

```diff
@@ -77,6 +77,12 @@ BOOST_FIXTURE_TEST_CASE(_dataset_manager_general, CompareCTypes) {
     dset_id_t ds_id;
     ds_id.set_from_string(line);
 
+    // Add a meaningless state in here. This is designed to trigger a failure
+    // case in older versions where if the local manager already knew about a
+    // state, it didn't bother checking to see if it knew about earlier states
+    //auto state_id = dm.create_state<flagState>("flag_id1").first;
+    //ds_id = dm.add_dataset(state_id, ds_id);
```

Comment:
Let's keep these 2 lines, though

Reply:
Oh, yeah. I struggled to get this to work. I think it may have been to do with the unclean exit problems. I'll try running again.